lib/components/primitive-components/SchematicBox.ts

```diff
@@ -112,5 +110,126 @@ export class SchematicBox extends PrimitiveComponent<typeof schematicBoxProps> {
       is_dashed: props.strokeStyle === "dashed",
       schematic_component_id: "",
     })
+
+    if (props.title) {
+      const isInside = props.titleInside ?? false
+
+      const getAnchorPosition = (
+        anchor: string,
+      ): {
+        x: number
+        y: number
+        textAnchor:
+          | "center"
+          | "top_left"
+          | "top_center"
+          | "top_right"
+          | "center_left"
+          | "center"
+          | "center_right"
+          | "bottom_left"
+          | "bottom_center"
+          | "bottom_right"
+      } => {
+        switch (anchor) {
+          case "top_left":
+            return {
+              x: x,
+              y: y + height,
+              textAnchor: isInside ? "top_left" : "bottom_left",
+            }
+          case "top_center":
+            return {
+              x: x + width / 2,
+              y: y + height,
+              textAnchor: isInside ? "top_center" : "bottom_center",
+            }
+          case "top_right":
+            return {
+              x: x + width,
+              y: y + height,
+              textAnchor: isInside ? "top_right" : "bottom_right",
+            }
+          case "center_left":
+            return {
+              x: x,
+              y: y + height / 2,
+              textAnchor: isInside ? "center_left" : "center_right",
+            }
+          case "center":
+            return { x: x + width / 2, y: y + height / 2, textAnchor: "center" }
+          case "center_right":
+            return {
+              x: x + width,
+              y: y + height / 2,
+              textAnchor: isInside ? "center_right" : "center_left",
+            }
+          case "bottom_left":
+            return {
+              x: x,
+              y: y,
+              textAnchor: isInside ? "bottom_left" : "top_left",
+            }
+          case "bottom_center":
+            return {
+              x: x + width / 2,
+              y: y,
+              textAnchor: isInside ? "bottom_center" : "top_center",
+            }
+          case "bottom_right":
+            return {
+              x: x + width,
+              y: y,
+              textAnchor: isInside ? "bottom_right" : "top_right",
+            }
+          default:
+            return { x: x + width / 2, y: y + height, textAnchor: "center" } // fallback: bottom-center
+        }
+      }
+
+      const anchor = props.titleAlignment ?? "bottom_center"
+      const anchorPos = getAnchorPosition(anchor)
+      let titleOffsetY: number
+      let titleOffsetX: number
+      const textAnchor = anchorPos.textAnchor
+      if (isInside) {
+        titleOffsetY = anchor.includes("top")
+          ? -0.05 * height
```

Comment:
Don't use percentage offsets, it looks terrible as things scale up.

Use fixed-margins e.g. 0.1mm etc.